### PR TITLE
Add a simple command level retry for case by case flaky test

### PR DESCRIPTION
**Pull request by wli3**

### CODE REVIEW

**Comment by wli3** [comment]
https://github.com/dotnet/runtime/issues/47431

File: src/Tests/Microsoft.NET.TestFramework/Commands/TestCommand.cs [hunk 1]
```diff
@@ -83,7 +95,21 @@ public ProcessStartInfo GetProcessStartInfo(params string[] args)
...
         {
             IEnumerable<string> enumerableArgs = args;
-            return Execute(enumerableArgs);
+
+            if (RetryCommand)
+            {
+                return ExecuteWithRetry(
+                    action: () => Execute(enumerableArgs),
+                    isSuccess: (result) => result.ExitCode == 0,
+                    maxRetryCount: 2,
```
Comment (joeloff) [inline src/Tests/Microsoft.NET.TestFramework/Commands/TestCommand.cs:104]: Is there an overload that allows setting the max retry count to a different value?

File: src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToPublishASingleFileApp.cs [hunk 2]
```diff
@@ -528,15 +528,19 @@ public void It_runs_single_file_apps(string targetFramework, bool selfContained,
             var testAsset = _testAssetsManager.CreateTestProject(testProject);
```
Comment (dsplaisted) [inline src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToPublishASingleFileApp.cs:528]:
```suggestion
            var testAsset = _testAssetsManager.CreateTestProject(testProject, identifier: targetFramework + "_" + selfContained + "_" + bundleOption);
```

The retry is a good idea, but I think that adding the appropriate identifier here will likely stop this test from being flaky so we won't need it.

**Review by sfoslund — APPROVED** [review]
Great idea! I agree with Jacques that a retry count parameter would be nice.

**Comment by wli3** [comment]
Let's try Daniel's fix first

File: src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToPublishASingleFileApp.cs (hunk 2 above)
Comment (wli3) [reply src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToPublishASingleFileApp.cs:528]: https://github.com/dotnet/sdk/pull/15571

**Comment by dsplaisted** [comment]
@wli3 Do we have a PR with my suggestion?

**Pull request closed without merging**